mm2src/mm2_main/tests/mm2_tests/tendermint_tests.rs

```diff
@@ -184,3 +199,164 @@ fn test_tendermint_token_activation_and_withdraw() {
     let send_raw_tx = block_on(send_raw_transaction(&mm, token, &tx_details.tx_hex));
     println!("Send raw tx {}", json::to_string(&send_raw_tx).unwrap());
 }
+
+#[test]
+fn test_tendermint_tx_history() { block_on(tendermint_tx_history()); }
+
+pub async fn tendermint_tx_history() {
+    const TEST_SEED: &str = "iris test seed3";
+    const TEST_ADDRESS: &str = "iaa17d8hndl72u2mzae8rymn003hxfpehcslnqcrcd";
+    const TEST_ADDRESS2: &str = "iaa1e0rx87mdj79zejewuc4jg7ql9ud2286g2us8f2";
+    const TX_FINISHED_LOG: &str = "Tx history fetching finished for tendermint.";
+
+    let coins = json!([iris_testnet_conf(), iris_nimda_testnet_conf()]);
+    let platform_coin = coins[0]["coin"].as_str().unwrap();
+    let token = coins[1]["coin"].as_str().unwrap();
+
+    let conf = Mm2TestConf::seednode(TEST_SEED, &coins);
+    let mut mm = MarketMakerIt::start_async(conf.conf, conf.rpc_password, None)
+        .await
+        .unwrap();
+
+    let activation_res = enable_tendermint(&mm, platform_coin, &[token], IRIS_TESTNET_RPC_URLS, true).await;
+    println!("Activation with assets {}", json::to_string(&activation_res).unwrap());
+
+    if let Err(_) = mm.wait_for_log(900., |log| log.contains(TX_FINISHED_LOG)).await {
+        println!("{}", mm.log_as_utf8().unwrap());
+        assert!(false, "Tx history didn't finish which is not expected");
+    }
+
+    let first_tx_history_response = get_tendermint_my_tx_history(&mm, platform_coin, 1, 1).await;
+    let first_total_txs = first_tx_history_response["result"]["total"].as_u64().unwrap();
+    assert!(first_total_txs > 0);
+
+    // self platform coin transfer
+    let withdraw_self = withdraw_v1(&mm, platform_coin, TEST_ADDRESS, "0.01").await;
+    let withdraw_self_tx_details: TransactionDetails = json::from_value(withdraw_self).unwrap();
+    let raw_tx_self = send_raw_transaction(&mm, platform_coin, &withdraw_self_tx_details.tx_hex).await;
+    let self_tx_hash = raw_tx_self["tx_hash"].as_str().unwrap();
```

Comment:
Performing withdraw will continuously increase the number of transactions on the address. It will make the test running longer and longer to fetch infinitely increasing history. With BCH and SLP, we use the address with pre-generated history that doesn't change.

Also, the test doesn't cover HTLC transactions happened during swap. I said "preferably with swaps" in https://github.com/KomodoPlatform/atomicDEX-API/pull/1526#pullrequestreview-1191484967, though. I think I should have mentioned it as mandatory change 🙂 

And we should also cover the case when multiple assets are transferred in a single transaction.

Reply:
> Also, the test doesn't cover HTLC transactions happened during swap. I said "preferably with swaps" in https://github.com/KomodoPlatform/atomicDEX-API/pull/1526#pullrequestreview-1191484967, though. I think I should have mentioned it as mandatory change slightly_smiling_face

This test implemented when swaps were failing due to regex bug that you found sometime ago. So I kind a had to imlement it this way.

> Performing withdraw will continuously increase the number of transactions on the address. It will make the test running longer and longer to fetch infinitely increasing history. With BCH and SLP, we use the address with pre-generated history that doesn't change.

> And we should also cover the case when multiple assets are transferred in a single transaction.

Will check them